### DISCUSSION TITLE
TensorFlow: How do I enable AMD GPU?

### DISCUSSION THREAD

**Discussion by wdckwrth** [photoprism/photoprism#2020] [Q&A] [answered]
I am interested in offloading the TF work in PP to an AMD GPU. I think it is possible but I am having trouble getting it set up. 

I am running PhotoPrism 220121-2b4c8e1f-Linux-x86_64 in a docker container. 

I have added devices to the docker-compose:

`    devices:
      - /dev/dri/renderD128:/dev/dri/renderD128
      - /dev/dri/card0:/dev/dri/card0`

My card is a Cape Verde XT [Radeon HD 7770/8760 / R7 250X]. I think it is configured correctly. I can run `radeontop` and it is recognized by the OS and inside the container.

I have tried adding PHOTOPRISM_INIT with tensorflow-amd64-avx or tensorflow-amd64-cpu. I can see them being added to `/tmp` but I do not see the GPU being used. If I add tensorflow-amd64-avx2 PP crashes on start. 

Any ideas? Thanks.

**Comment by lastzero** [comment, marked as answer]
From what I know, AMD hardware acceleration is not supported by TensorFlow. Maybe they have added this since I last checked, so do your own research :+1:

**Reply by wdckwrth** [reply]
From what I have been able to dig up it seems like TensorFlow is supported on AMD hardware via [ROCm](https://www.amd.com/en/technologies/infinity-hub/tensorflow). There are [specific chip versions](https://community.amd.com/t5/knowledge-base/amd-rocm-hardware-and-software-support-document/ta-p/489937) required and additional libraries necessary. 

My card, a GFX6, is not supported so I think I am at a dead end. Folks with GFX7 or newer might be able to test.

**Reply by afrojewelz** [reply]
if your AMD GPU support macOS, i suggest install run photoprism on macOS utillize apple's Apple Video Toolbox api to meet requirement of transcode, as for gpu compute ,only option is opencl on macOS if metal api is not supported

**Comment by ssorgatem** [comment]
Any update on this?
It seems quite easy to implement as an option in PHOTOPRISM_INIT (e.g. tensorflow-rocm instead of tensorflow), besides the necessary docker/compose config changes to allow access to the GPU (documented here: https://rocm.docs.amd.com/projects/install-on-linux/en/latest/how-to/3rd-party/tensorflow-install.html)

**Comment by ssorgatem** [comment]
Hmm, after some testing, it seems the Python version that comes in the Ubuntu image used for the docker images is too new (3.12), with the latest version supported by AMD being 3.10... so it may be a bit more work than it should. At least, until they release tensorflow-rocm for Python 3.12... as long as that happens before Photoprism moves to a newer Python.

**Reply by pmferraro** [reply]
Any update??
